api/src/main/java/com/capitalone/dashboard/webhook/github/GitHubPullRequestV3.java

```diff
@@ -0,0 +1,503 @@
+import org.springframework.data.querydsl.QueryDslPredicateExecutor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestClientException;
+
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+
+public class GitHubPullRequestV3 extends GitHubV3 {
+    private static final Log LOG = LogFactory.getLog(GitHubPullRequestV3.class);
+
+    private final GitRequestRepository gitRequestRepository;
+    private final GitHubRepoRepository gitHubRepoRepository;
+    private final CommitRepository commitRepository;
+
+    private Map<String, String> ldapDNMap = new HashMap<>();
+
+    public GitHubPullRequestV3(CollectorService collectorService,
+                               RestClient restClient,
+                               GitRequestRepository gitRequestRepository,
+                               CommitRepository commitRepository,
+                               GitHubRepoRepository gitHubRepoRepository,
+                               ApiSettings apiSettings) {
+        super(collectorService, restClient, apiSettings);
+
+        this.gitRequestRepository = gitRequestRepository;
+        this.gitHubRepoRepository = gitHubRepoRepository;
+        this.commitRepository = commitRepository;
+    }
+
+    @Override
+    public QueryDslPredicateExecutor<GitHubRepo> getGitHubRepoRepository() { return this.gitHubRepoRepository; }
+
+    @Override
+    public String process(JSONObject prJsonObject) throws MalformedURLException, HygieiaException, ParseException {
+        Object pullRequestObject = restClient.getAsObject(prJsonObject, "pull_request");
+        if (pullRequestObject == null) return "Pull Request Data Not Available";
+
+        int prNumber = restClient.getInteger(pullRequestObject,"number");
+        if (prNumber == 0) return "Pull Request Number Not Available";
+
+        Object repoMap = prJsonObject.get("repository");
+        if (repoMap == null) { return "Repository Data Not Available"; }
+
+        String repoUrl = restClient.getString(repoMap, "html_url");
+        GitHubParsed gitHubParsed = new GitHubParsed(repoUrl);
+
+        boolean isPrivate = restClient.getBoolean(repoMap, "private");
+
+        JSONObject postBody = buildGraphQLQuery(gitHubParsed, pullRequestObject);
+
+        if (postBody == null) { return "No Commits found on the PR. Returning ...";}
+
+        GitHubWebHookSettings gitHubWebHookSettings = parseAsGitHubWebHook(apiSettings.getGitHubWebHook());
+        String gitHubWebHookToken = null;
+        if (gitHubWebHookSettings != null) {
+            gitHubWebHookToken = gitHubWebHookSettings.getToken();
+        }
+
+        long start = System.currentTimeMillis();
+
+        String repoToken = getRepositoryToken(repoUrl);
+
+        long end = System.currentTimeMillis();
+        LOG.debug("Time to make gitHubRepoRepository call = "+(end-start));
```

Comment:
Can you change the verbiage to say gitHubRepoRepository call to fetch the repo token

Reply:
Ok, I have updated this, thanks.